OpenRA.Game/GameRules/Ruleset.cs

```diff
@@ -227,10 +227,13 @@ static bool AnyFlaggedTraits(ModData modData, List<MiniYamlNode> actors)
 					{
 						var traitName = traitNode.Key.Split('@')[0];
 						var traitType = modData.ObjectCreator.FindType(traitName + "Info");
-						if (traitType.GetInterface("ILobbyCustomRulesIgnore") == null)
+						if (traitType != null && traitType.GetInterface("ILobbyCustomRulesIgnore") == null)
 							return true;
 					}
-					catch { }
+					catch (Exception ex)
+					{
+						Log.Write("debug", ex.ToString());
```

Comment:
Consider `"Error in AnyFlaggedTraits\r\n" + ex.ToString()` so there's some context.

Reply:
Fair point for the logging.

 (Although it does potentially suggest a pattern of an identifier param paired with an exception param, given how many codepaths simply log exception strings)